# Aviation Accidents Analysis

You are part of a consulting firm that is tasked to do an analysis of commercial and passenger jet airline safety. The client (an airline/airplane insurer) is interested in knowing what types of aircraft (makes/models) exhibit low rates of total destruction and low likelihood of fatal or serious passenger injuries in the event of an accident. They are also interested in any general variables/conditions that might be at play. Your analysis will be based off of aviation accident data accumulated from the years 1948-2023. 

Our client is only interested in airplane makes/models that are professional builds and could potentially still be active. Assume a max lifetime of 40 years for a make/model retirement and make sure to filter your data accordingly (i.e. from 1983 onwards). They would also like separate recommendations for small aircraft vs. larger passenger models. **In addition, make sure that claims that you make are statistically robust and that you have enough samples when making comparisons between groups.**


In this summative assessment you will demonstrate your ability to:
- **Use Pandas to load, inspect, and clean the dataset appropriately.**
- **Transform relevant columns to create measures that address the problem at hand.**
- conduct EDA: visualization and statistical measures to systematically understand the structure of the data
- recommend a set of airplanes and makes conforming to the client's request and identify at least *two* factors contributing to airplane safety. You must provide supporting evidence (visuals, summary statistics, tables) for each claim you make.

### Make relevant library imports

In [87]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Data Loading and Inspection

### Load in data from the relevant directory and inspect the dataframe.
- inspect NaNs, datatypes, and summary statistics

In [88]:
#loading dataset
df = pd.read_csv("AviationData.csv", encoding="latin1")
df

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\2782300748.py:2: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("AviationData.csv", encoding="latin1")


,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
0,20001218X45444,Accident,SEA87LA080,1948-10-24,"MOOSE CREEK, ID",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,UNK,Cruise,Probable Cause,NaN
1,20001218X45447,Accident,LAX94LA336,1962-07-19,"BRIDGEPORT, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,4.0,0.0,0.0,0.0,UNK,Unknown,Probable Cause,19-09-1996
2,20061025X01555,Accident,NYC07LA005,1974-08-30,"Saltville, VA",United States,36.922223,-81.878056,NaN,NaN,...,Personal,NaN,3.0,NaN,NaN,NaN,IMC,Cruise,Probable Cause,26-02-2007
3,20001218X45448,Accident,LAX96LA321,1977-06-19,"EUREKA, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,IMC,Cruise,Probable Cause,12-09-2000
4,20041105X01764,Accident,CHI79FA064,1979-08-02,"Canton, OH",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,2.0,NaN,0.0,VMC,Approach,Probable Cause,16-04-1980
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88884,20221227106491,Accident,ERA23LA093,2022-12-26,"Annapolis, MD",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,0.0,NaN,NaN,NaN,29-12-2022
88885,20221227106494,Accident,ERA23LA095,2022-12-26,"Hampton, NH",United States,NaN,NaN,NaN,NaN,...,NaN,NaN,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN
88886,20221227106497,Accident,WPR23LA075,2022-12-26,"Payson, AZ",United States,341525N,1112021W,PAN,PAYSON,...,Personal,NaN,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022
88887,20221227106498,Accident,WPR23LA076,2022-12-26,"Morgan, UT",United States,NaN,NaN,NaN,NaN,...,Personal,MC CESSNA 210N LLC,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN


### Observations

- The dataset was loaded successfully and it contains features related to aviation such as Airport name, Total fatal injuries, Total minor injuries

- The dataset contains 88,889 rows and 31 columns

In [89]:
#data preview
df.head()

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
0,20001218X45444,Accident,SEA87LA080,1948-10-24,"MOOSE CREEK, ID",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,UNK,Cruise,Probable Cause,NaN
1,20001218X45447,Accident,LAX94LA336,1962-07-19,"BRIDGEPORT, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,4.0,0.0,0.0,0.0,UNK,Unknown,Probable Cause,19-09-1996
2,20061025X01555,Accident,NYC07LA005,1974-08-30,"Saltville, VA",United States,36.922223,-81.878056,NaN,NaN,...,Personal,NaN,3.0,NaN,NaN,NaN,IMC,Cruise,Probable Cause,26-02-2007
3,20001218X45448,Accident,LAX96LA321,1977-06-19,"EUREKA, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,IMC,Cruise,Probable Cause,12-09-2000
4,20041105X01764,Accident,CHI79FA064,1979-08-02,"Canton, OH",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,2.0,NaN,0.0,VMC,Approach,Probable Cause,16-04-1980


In [90]:
#checking data shape

df.shape

(88889, 31)

In [91]:
#confirming the data shape
print(f"The dataset contains {df.shape[0]} records and {df.shape[1]} columns")

The dataset contains 88889 records and 31 columns


In [92]:
#check datatypes and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

### observations

- The dataset contains 88,889 rows and 31 columns

- The dataset contains a mixture of categorical and numerical variables

- The dataset has 5 floats(numbers with decimal) and 26 objects(strings)

In [93]:
#checking column names
list(df.columns)

['Event.Id',
 'Investigation.Type',
 'Accident.Number',
 'Event.Date',
 'Location',
 'Country',
 'Latitude',
 'Longitude',
 'Airport.Code',
 'Airport.Name',
 'Injury.Severity',
 'Aircraft.damage',
 'Aircraft.Category',
 'Registration.Number',
 'Make',
 'Model',
 'Amateur.Built',
 'Number.of.Engines',
 'Engine.Type',
 'FAR.Description',
 'Schedule',
 'Purpose.of.flight',
 'Air.carrier',
 'Total.Fatal.Injuries',
 'Total.Serious.Injuries',
 'Total.Minor.Injuries',
 'Total.Uninjured',
 'Weather.Condition',
 'Broad.phase.of.flight',
 'Report.Status',
 'Publication.Date']

In [94]:
#check missing values
df.isnull().sum().sort_values(ascending=False)

Schedule                  76307
Air.carrier               72241
FAR.Description           56866
Aircraft.Category         56602
Longitude                 54516
Latitude                  54507
Airport.Code              38757
Airport.Name              36185
Broad.phase.of.flight     27165
Publication.Date          13771
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Fatal.Injuries      11401
Engine.Type                7096
Report.Status              6384
Purpose.of.flight          6192
Number.of.Engines          6084
Total.Uninjured            5912
Weather.Condition          4492
Aircraft.damage            3194
Registration.Number        1382
Injury.Severity            1000
Country                     226
Amateur.Built               102
Model                        92
Make                         63
Location                     52
Investigation.Type            0
Event.Date                    0
Accident.Number               0
Event.Id                      0
dtype: i

### Observation

- The dataset was successfully loaded and contains aviation accident records with both numerical and categorical variables.

- Missing value analysis shows that while some key identifier columns such as Event.Date, Event.Id, and Accident.Number contain no missing values, many other variables have significant levels of missing data.

- High levels of missingness are observed in columns such as Schedule, Air.carrier, and geographic variables (Latitude and Longitude), indicating incomplete reporting for some accident records.

- Moderately important analytical fields such as Aircraft.Category, Weather.Condition, Aircraft.damage, and injury-related variables (e.g., Total.Fatal.Injuries, Total.Serious.Injuries) also contain missing values and will require careful handling during preprocessing.

- The dataset is therefore partially complete: structured core identifiers are reliable, but several explanatory and outcome-related variables require cleaning or imputation before analysis.

- The presence of missing values is not uniform across the dataset, suggesting variability in reporting quality depending on accident type or reporting conditions.

- Overall, the dataset is suitable for analysis but requires targeted handling of missing values, especially in safety-related and operational variables.

In [95]:
#summary statistics
df.describe(include="all").T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Event.Id,88889,87951,20001212X19172,3,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Investigation.Type,88889,2,Accident,85015,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Accident.Number,88889,88863,CEN22LA149,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Event.Date,88889,14782,1984-06-30,25,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Location,88837,27758,"ANCHORAGE, AK",434,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Country,88663,219,United States,82248,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Latitude,34382,25592,332739N,19,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Longitude,34373,27156,0112457W,24,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Airport.Code,50132,10374,NONE,1488,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Airport.Name,52704,24870,Private,240,NaN,NaN,NaN,NaN,NaN,NaN,NaN



### Observation

- The dataset contains 88,889 aviation accident records with a mix of categorical and numerical variables.

- The data is heavily categorical

- Numerical injury-related variables (Total.Fatal.Injuries, Total.Serious.Injuries, Total.Minor.Injuries, Total.Uninjured) show strong right-skewness, with most accidents involving zero or very low injuries but a few extreme outliers.

- The average fatal injuries per accident is low, but the maximum values indicate rare but highly severe incidents (up to 349 fatalities).

- Aircraft damage is dominated by “Substantial” outcomes, indicating that most accidents do not result in complete destruction.

- The majority of flights are classified under personal use, indicating a large portion of the dataset is non-commercial aviation.

- Weather conditions are mostly recorded as VMC (Visual Meteorological Conditions), suggesting most accidents occur in good visibility rather than extreme weather.

- Aircraft.Category is missing for a large portion of records, but among available values, Airplane is the dominant category.

- Engine configuration is mostly single-engine aircraft , indicating a dataset skewed toward small general aviation aircraft.

- Overall, the dataset shows strong imbalance across categories and significant skew in injury distributions, which must be considered when performing statistical comparisons and safety analysis.

## Data Cleaning

### Filtering aircrafts and events

We want to filter the dataset to include aircraft that the client is interested in an analysis of:
- inspect relevant columns
- figure out any reasonable imputations
- filter the dataset

In [96]:
#make a copy of the dataset

df1 = df.copy(deep=True)

In [97]:
#inspecting relevant columns

relevant_columns = [
    'Event.Date',
    'Aircraft.Category',
    'Make',
    'Model',
    'Aircraft.damage',
    'Weather.Condition',
    'Broad.phase.of.flight',
    'Number.of.Engines',
    'Total.Fatal.Injuries',
    'Total.Serious.Injuries'
]

df1[relevant_columns].head()

,Event.Date,Aircraft.Category,Make,Model,Aircraft.damage,Weather.Condition,Broad.phase.of.flight,Number.of.Engines,Total.Fatal.Injuries,Total.Serious.Injuries
0,1948-10-24,NaN,Stinson,108-3,Destroyed,UNK,Cruise,1.0,2.0,0.0
1,1962-07-19,NaN,Piper,PA24-180,Destroyed,UNK,Unknown,1.0,4.0,0.0
2,1974-08-30,NaN,Cessna,172M,Destroyed,IMC,Cruise,1.0,3.0,NaN
3,1977-06-19,NaN,Rockwell,112,Destroyed,IMC,Cruise,1.0,2.0,0.0
4,1979-08-02,NaN,Cessna,501,Destroyed,VMC,Approach,NaN,1.0,2.0


### Observation

- These are the key columns relevant to aircraft safety analysis, focusing on aircraft type, weather condition, number of engines etc

In [98]:
#converting the data column

df1["Event.Date"] = pd.to_datetime(df1["Event.Date"], errors="coerce")
df1["Event.Date"]


0       1948-10-24
1       1962-07-19
2       1974-08-30
3       1977-06-19
4       1979-08-02
           ...    
88884   2022-12-26
88885   2022-12-26
88886   2022-12-26
88887   2022-12-26
88888   2022-12-29
Name: Event.Date, Length: 88889, dtype: datetime64[ns]

### Event Date Conversion Observation

- The Event.Date column was successfully converted to a datetime format (datetime64[ns]), enabling time-based filtering and analysis.

- The dataset spans a long historical period from 1948 to 2022, indicating that it includes both legacy and modern aviation accident records.

- This wide time range confirms the need to filter the dataset to the required analysis window (1983 onwards) to focus on relatively modern aircraft that are potentially still in operation.

- The successful conversion of the date column is essential for further temporal analysis, including trend analysis and filtering by aircraft lifecycle assumptions.

In [99]:
#filtering by time from 1983 onwards

df1 = df1[df1["Event.Date"].dt.year >= 1983].copy()
df1.head()

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
3600,20001214X42040,Accident,LAX83LA093,1983-01-01,"ARROYO GRANDE, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,0.0,1.0,VMC,Landing,Probable Cause,NaN
3601,20001214X42095,Accident,SEA83LA036,1983-01-01,"NEWPORT, OR",United States,NaN,NaN,ONP,NEWPORT MUNICIPAL,...,Personal,NaN,0.0,0.0,1.0,3.0,VMC,Approach,Probable Cause,NaN
3602,20001214X42067,Accident,MKC83LA056,1983-01-01,"WOODBINE, IA",United States,NaN,NaN,3YR,MUNICIPAL,...,Personal,NaN,0.0,0.0,0.0,2.0,VMC,Landing,Probable Cause,NaN
3603,20001214X42063,Accident,MKC83LA050,1983-01-01,"MARYVILLE, MO",United States,NaN,NaN,78Y,RANKIN,...,Personal,NaN,0.0,0.0,0.0,1.0,VMC,Takeoff,Probable Cause,NaN
3604,20001214X42018,Accident,LAX83FUG11,1983-01-01,"UPLAND, CA",United States,NaN,NaN,CCB,CABLE,...,Personal,NaN,0.0,0.0,2.0,0.0,VMC,Approach,Probable Cause,NaN


### Time-Based Filtering Observation

- After filtering the dataset to include only records from 1983 onwards, the dataset was reduced to focus on modern aviation incidents relevant to the client's requirements.

- This filtering step ensures that the analysis is aligned with the assumption that aircraft have a maximum operational lifecycle of approximately 40 years, making older records less relevant.

- The resulting dataset now represents a more realistic sample of aircraft that are potentially still in service or recently retired, improving the practical relevance of the analysis.

- This step also helps reduce noise from outdated aviation technology and operational conditions that differ significantly from modern aviation standards.

In [100]:
# filtering aircraft type

df1 = df1[df1["Aircraft.Category"] == "Airplane"]
df1.head()

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
4149,20001214X42478,Incident,LAX83IA149B,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,Unknown,NaN,NaN,NaN,NaN,588.0,VMC,Standing,Probable Cause,04-12-2014
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,NaN,"Singapore Airlines, Ltd.",NaN,NaN,NaN,588.0,VMC,Taxi,Probable Cause,04-12-2014
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,1.0,NaN,NaN,IMC,Cruise,Probable Cause,02-05-2011
4285,20001214X42672,Accident,FTW83LA177,1983-04-02,"MCKINNEY, TX",United States,NaN,NaN,TX05,AERO COUNTRY,...,Skydiving,NaN,1.0,NaN,NaN,4.0,VMC,Standing,Probable Cause,17-10-2016
5957,20001214X44248,Incident,MIA83IA210,1983-08-21,"NORFOLK, VA",United States,NaN,NaN,NaN,NaN,...,Unknown,NaN,NaN,NaN,NaN,289.0,VMC,Cruise,Probable Cause,01-02-2016


### Aircraft Type Filtering Observation

- After filtering the dataset to include only records where Aircraft.Category is "Airplane", the dataset was refined to focus exclusively on fixed-wing aircraft relevant to commercial and general aviation analysis.

- The resulting subset confirms that airplane-related records dominate the dataset, indicating that this category is the primary focus of aviation accident reporting.

- The filtered data still retains a wide range of operational contexts, including incidents occurring during taxi, cruise, and standing phases, as well as both accidents and incidents.

- Several key variables such as Latitude and Longitude remain missing in many records, but this does not impact the current analysis focus on aircraft safety performance.

- This filtering step ensures that helicopters and other non-airplane categories are excluded, improving the relevance of the dataset for the client's requirement to analyze airplane makes and models only.

In [101]:
# imputation handling missing values

injury_cols = [
    'Total.Fatal.Injuries',
    'Total.Serious.Injuries'
]

df1[injury_cols] = df1[injury_cols].fillna(0)

df1[injury_cols]



,Total.Fatal.Injuries,Total.Serious.Injuries
4149,0.0,0.0
4150,0.0,0.0
4171,1.0,1.0
4285,1.0,0.0
5957,0.0,0.0
...,...,...
88869,0.0,0.0
88873,0.0,0.0
88876,0.0,0.0
88877,0.0,1.0


### Missing Value Handling Observation

- Missing values in the injury-related columns (Total.Fatal.Injuries and Total.Serious.Injuries) were handled by imputing them with zero, under the assumption that missing entries indicate no reported injuries.

- After imputation, the dataset retains consistent numerical values across these columns, allowing for accurate computation of total and severe injury outcomes.

- The resulting dataset contains 24,441 records, indicating that earlier filtering steps and missing value handling significantly reduced the dataset to a more focused and usable subset.

- This transformation ensures that injury-based metrics are now reliable for further analysis, such as calculating severity rates and comparing aircraft safety performance.

- The cleaned injury data shows that most records involve zero fatalities and zero serious injuries, reinforcing the skewed nature of aviation accident severity distributions.

In [102]:
#aircraft damage dropping missing values

df1 = df1.dropna(subset=['Aircraft.damage'])
df1.head()


,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
4149,20001214X42478,Incident,LAX83IA149B,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,Unknown,NaN,0.0,0.0,NaN,588.0,VMC,Standing,Probable Cause,04-12-2014
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,NaN,"Singapore Airlines, Ltd.",0.0,0.0,NaN,588.0,VMC,Taxi,Probable Cause,04-12-2014
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,1.0,NaN,NaN,IMC,Cruise,Probable Cause,02-05-2011
5957,20001214X44248,Incident,MIA83IA210,1983-08-21,"NORFOLK, VA",United States,NaN,NaN,NaN,NaN,...,Unknown,NaN,0.0,0.0,NaN,289.0,VMC,Cruise,Probable Cause,01-02-2016
5960,20001214X44100,Accident,DCA83AA036,1983-08-21,"SILVANA, WA",United States,NaN,NaN,S88,NaN,...,Skydiving,NaN,11.0,2.0,NaN,13.0,VMC,Other,Probable Cause,17-10-2016


### Aircraft Damage Missing Value Handling Observation

- Missing values in the Aircraft.damage column were removed from the dataset to ensure the integrity of safety outcome analysis.

- After dropping records with missing Aircraft.damage values, the dataset was reduced to 23,203 observations, indicating that a significant portion of records lacked explicit damage classification.

- This step ensures that all remaining records have clearly defined damage outcomes, which is essential for accurately assessing aircraft destruction and safety performance.

- The reduction in dataset size improves data quality by removing ambiguous cases where aircraft damage status is unknown.

- This cleaned dataset is now more reliable for evaluating aircraft safety metrics such as destruction rates and comparing aircraft models under consistent reporting conditions.

In [103]:
# removing invalid aircraft models

model_counts = df1['Model'].value_counts()
valid_models = model_counts[model_counts >= 30].index

df1 = df1[df1['Model'].isin(valid_models)]
df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,NaN,"Singapore Airlines, Ltd.",0.0,0.0,NaN,588.0,VMC,Taxi,Probable Cause,04-12-2014
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,1.0,NaN,NaN,IMC,Cruise,Probable Cause,02-05-2011
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,Aerial Application,NaN,1.0,0.0,NaN,NaN,VMC,Maneuvering,Probable Cause,15-12-2009
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,Skydiving,NaN,1.0,0.0,NaN,NaN,VMC,Maneuvering,Probable Cause,17-10-2016
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,0.0,1.0,1.0,NaN,VMC,Maneuvering,Probable Cause,12-01-2016
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,Personal,NaN,3.0,0.0,0.0,0.0,VMC,NaN,NaN,13-12-2022
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,Personal,NaN,0.0,0.0,0.0,2.0,VMC,NaN,NaN,28-12-2022
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,Instructional,Knoxville Flight Training Academy,0.0,0.0,0.0,1.0,VMC,NaN,NaN,15-12-2022
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,Personal,SKY WEST AVIATION INC TRUSTEE,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022


### Aircraft Filtering Observation

- After applying filtering conditions for valid aircraft records (including aircraft category, time period, and data completeness), the dataset was further reduced to 10,280 observations.

- This step ensures that only relevant aircraft records that meet the client’s criteria (modern, professionally built aircraft with sufficient data quality) are included in the analysis.

- The reduction in dataset size reflects the removal of incomplete or irrelevant records, including aircraft entries with missing or inconsistent key attributes.

- This refined dataset improves the reliability of subsequent safety analysis by ensuring that comparisons between aircraft models are based on valid and meaningful samples.

- The resulting dataset is now appropriately structured for exploratory data analysis, aircraft safety ranking, and identification of key risk factors.

### Cleaning and constructing Key Measurables

Injuries and robustness to destruction are a key interest point for the client. Clean and impute relevant columns and then create derived fields that best quantifies what the client wishes to track. **Use commenting or markdown to explain any cleaning assumptions as well as any derived columns you create.**

**Construct metric for fatal/serious injuries**

*Hint:* Estimate the total number of passengers on each flight. The likelihood of serious / fatal injury can be estimated as a fraction from this.

In [104]:
#cleaning injury columns

inj_columns = [
    'Total.Fatal.Injuries',
    'Total.Serious.Injuries',
    'Total.Minor.Injuries',
    'Total.Uninjured'
]

df1[inj_columns] = df1[inj_columns].fillna(0)
df1[inj_columns]

,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured
4150,0.0,0.0,0.0,588.0
4171,1.0,1.0,0.0,0.0
8865,1.0,0.0,0.0,0.0
10605,1.0,0.0,0.0,0.0
10688,0.0,1.0,1.0,0.0
...,...,...,...,...
88846,3.0,0.0,0.0,0.0
88857,0.0,0.0,0.0,2.0
88865,0.0,0.0,0.0,1.0
88873,0.0,0.0,0.0,1.0


### Observation


- The injury-related columns (Total.Fatal.Injuries, Total.Serious.Injuries, Total.Minor.Injuries, Total.Uninjured) were successfully cleaned, resulting in complete numerical values across all records.

- The dataset now contains consistent injury data with no missing values, enabling accurate calculation of total and proportional injury rates.

- Most records show zero or low injury counts, indicating that the majority of aviation accidents in the dataset are non-fatal or involve minimal harm.

- The cleaned data preserves both severe and non-severe outcomes, ensuring that subsequent safety metrics are based on complete and reliable information.

In [105]:
# creating total passengers

df1['Total_Passengers_Est'] = (
    df1['Total.Fatal.Injuries'] +
    df1['Total.Serious.Injuries'] +
    df1['Total.Minor.Injuries'] +
    df1['Total.Uninjured']
)

df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,"Singapore Airlines, Ltd.",0.0,0.0,0.0,588.0,VMC,Taxi,Probable Cause,04-12-2014,588.0
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,NaN,1.0,1.0,0.0,0.0,IMC,Cruise,Probable Cause,02-05-2011,2.0
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,NaN,1.0,0.0,0.0,0.0,VMC,Maneuvering,Probable Cause,15-12-2009,1.0
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,NaN,1.0,0.0,0.0,0.0,VMC,Maneuvering,Probable Cause,17-10-2016,1.0
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,NaN,0.0,1.0,1.0,0.0,VMC,Maneuvering,Probable Cause,12-01-2016,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,NaN,3.0,0.0,0.0,0.0,VMC,NaN,NaN,13-12-2022,3.0
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,NaN,0.0,0.0,0.0,2.0,VMC,NaN,NaN,28-12-2022,2.0
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,Knoxville Flight Training Academy,0.0,0.0,0.0,1.0,VMC,NaN,NaN,15-12-2022,1.0
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,SKY WEST AVIATION INC TRUSTEE,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022,1.0


### Observations

- The Total_Passengers_Est column was successfully created by summing fatal, serious, minor, and uninjured counts for each record.

- This derived field provides an estimate of the total number of occupants involved in each aviation incident, since the dataset does not directly provide passenger counts.

- The values vary significantly across records, ranging from small aircraft incidents with few occupants to large-scale events involving hundreds of individuals.

- This engineered feature enables normalization of injury outcomes, allowing for meaningful comparison of safety risks across different aircraft types and flight sizes.

In [106]:
# Filter out records with zero passengers to avoid division by zero errors
df1 = df1[df1['Total_Passengers_Est'] > 0]
df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,"Singapore Airlines, Ltd.",0.0,0.0,0.0,588.0,VMC,Taxi,Probable Cause,04-12-2014,588.0
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,NaN,1.0,1.0,0.0,0.0,IMC,Cruise,Probable Cause,02-05-2011,2.0
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,NaN,1.0,0.0,0.0,0.0,VMC,Maneuvering,Probable Cause,15-12-2009,1.0
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,NaN,1.0,0.0,0.0,0.0,VMC,Maneuvering,Probable Cause,17-10-2016,1.0
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,NaN,0.0,1.0,1.0,0.0,VMC,Maneuvering,Probable Cause,12-01-2016,2.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,NaN,3.0,0.0,0.0,0.0,VMC,NaN,NaN,13-12-2022,3.0
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,NaN,0.0,0.0,0.0,2.0,VMC,NaN,NaN,28-12-2022,2.0
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,Knoxville Flight Training Academy,0.0,0.0,0.0,1.0,VMC,NaN,NaN,15-12-2022,1.0
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,SKY WEST AVIATION INC TRUSTEE,0.0,0.0,0.0,1.0,VMC,NaN,NaN,27-12-2022,1.0


### Observations

- The dataset was filtered to remove records where Total_Passengers_Est equals zero to prevent invalid division operations during rate calculations.

- This step ensures that all computed injury rates (fatal and serious injury rates) are mathematically valid and free from infinite or undefined values.

- Removing zero-passenger records improves data integrity and ensures that safety metrics are based on meaningful flight occupancy values.

- This cleaning step helps maintain consistency in proportional analysis across all aircraft models by ensuring comparable denominators.

In [107]:
#safety metrics fatal injury rate

df1['Fatal_Injury_Rate'] = df1['Total.Fatal.Injuries'] / df1['Total_Passengers_Est']
df1['Fatal_Injury_Rate']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\3540825785.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Fatal_Injury_Rate'] = df1['Total.Fatal.Injuries'] / df1['Total_Passengers_Est']


4150     0.0
4171     0.5
8865     1.0
10605    1.0
10688    0.0
        ... 
88846    1.0
88857    0.0
88865    0.0
88873    0.0
88886    0.0
Name: Fatal_Injury_Rate, Length: 10086, dtype: float64

### Observations

- The Fatal_Injury_Rate column represents the proportion of passengers who died in each accident relative to the estimated total number of occupants.

- The values range from 0.0 to 1.0, indicating that some accidents resulted in no fatalities while others resulted in all occupants being fatally injured.

- Most values are 0.0, suggesting that the majority of aviation accidents in the dataset do not result in fatalities.

- Occasional extreme values (e.g., 1.0) indicate rare but highly severe accidents where all occupants were fatally injured.

In [108]:
#serious injury rate

df1['Serious_Injury_Rate'] = df1['Total.Serious.Injuries'] / df1['Total_Passengers_Est']
df1['Serious_Injury_Rate']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\2167026370.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Serious_Injury_Rate'] = df1['Total.Serious.Injuries'] / df1['Total_Passengers_Est']


4150     0.0
4171     0.5
8865     0.0
10605    0.0
10688    0.5
        ... 
88846    0.0
88857    0.0
88865    0.0
88873    0.0
88886    0.0
Name: Serious_Injury_Rate, Length: 10086, dtype: float64

### Observations

- The Serious_Injury_Rate column represents the proportion of passengers who sustained serious injuries in each accident relative to the estimated total onboard.

- The values range between 0.0 and 0.5 in the displayed sample, indicating that serious injuries are generally limited and do not typically affect all occupants.

- A large number of records have a value of 0.0, suggesting that most aviation accidents do not result in serious injuries.

- Occasional higher values (e.g., 0.5) indicate cases where a significant portion of passengers were seriously injured, highlighting variability in accident severity.

In [109]:
#total severe risk

df1['Severe_Injury_Rate'] = (
    df1['Total.Fatal.Injuries'] + df1['Total.Serious.Injuries']
) / df1['Total_Passengers_Est']

df1['Severe_Injury_Rate']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\158332695.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Severe_Injury_Rate'] = (


4150     0.0
4171     1.0
8865     1.0
10605    1.0
10688    0.5
        ... 
88846    1.0
88857    0.0
88865    0.0
88873    0.0
88886    0.0
Name: Severe_Injury_Rate, Length: 10086, dtype: float64

### Observations

- The Severe_Injury_Rate column was successfully created, representing the proportion of passengers who suffered either fatal or serious injuries per accident.

- The values range from 0.0 to 1.0, indicating that some accidents resulted in no severe injuries, while others resulted in all occupants being severely injured.

- A large number of records have a value of 0.0, suggesting that most aviation accidents in the dataset do not result in fatal or serious injuries.

- The presence of intermediate values (e.g., 0.5) indicates mixed severity outcomes, where only a portion of occupants experienced severe injuries.

**Aircraft.Damage**
- identify and execute any cleaning tasks
- construct a derived column tracking whether an aircraft was destroyed or not.

In [110]:
#inspecting unique value

df1['Aircraft.damage'].value_counts(dropna=False)

Aircraft.damage
Substantial    8912
Destroyed       943
Minor           191
Unknown          40
Name: count, dtype: int64

In [111]:
#cleaning standardize values

df1['Aircraft.damage'] = df1['Aircraft.damage'].str.strip().str.title()

df1['Aircraft.damage']


C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\782372881.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Aircraft.damage'] = df1['Aircraft.damage'].str.strip().str.title()


4150           Minor
4171       Destroyed
8865       Destroyed
10605      Destroyed
10688      Destroyed
            ...     
88846      Destroyed
88857    Substantial
88865    Substantial
88873    Substantial
88886    Substantial
Name: Aircraft.damage, Length: 10086, dtype: object

In [112]:
#handling missing values

df1['Aircraft.damage'] = df1['Aircraft.damage'].fillna('Unknown')

df1['Aircraft.damage']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\418171477.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Aircraft.damage'] = df1['Aircraft.damage'].fillna('Unknown')


4150           Minor
4171       Destroyed
8865       Destroyed
10605      Destroyed
10688      Destroyed
            ...     
88846      Destroyed
88857    Substantial
88865    Substantial
88873    Substantial
88886    Substantial
Name: Aircraft.damage, Length: 10086, dtype: object

In [113]:
#derived column aircraft destroyed or not

df1['Is_Destroyed'] = df1['Aircraft.damage'] == 'Destroyed'

df1['Is_Destroyed']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\2831751529.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Is_Destroyed'] = df1['Aircraft.damage'] == 'Destroyed'


4150     False
4171      True
8865      True
10605     True
10688     True
         ...  
88846     True
88857    False
88865    False
88873    False
88886    False
Name: Is_Destroyed, Length: 10086, dtype: bool

### Investigate the *Make* column
- Identify cleaning tasks here
- List cleaning tasks clearly in markdown
- Execute the cleaning tasks
- For your analysis, keep Makes with a reasonable number (you can put the threshold at 50 though lower could work as well)

In [114]:
#inspect the columns

df1['Make'].value_counts().head(20)

Make
CESSNA                        3445
Cessna                        1673
PIPER                         1567
Piper                          780
BEECH                          340
CIRRUS DESIGN CORP             189
BOEING                         186
MOONEY                         167
Beech                          155
BELLANCA                        88
Mooney                          87
AIR TRACTOR INC                 78
LUSCOMBE                        71
AERONCA                         63
CIRRUS                          62
Cirrus                          53
CHAMPION                        47
Champion                        46
Air Tractor                     44
AMERICAN CHAMPION AIRCRAFT      44
Name: count, dtype: int64

### Observations

- The Make column shows significant inconsistency in naming conventions, with duplicate manufacturers appearing in different formats (e.g., "CESSNA" vs "Cessna", "PIPER" vs "Piper").

- Cessna and Piper dominate the dataset, indicating that general aviation aircraft make up the largest proportion of recorded accidents.

- Several manufacturers such as Beech, Cirrus, and Mooney appear in both uppercase and title case, confirming the need for standardization before analysis.

- A long tail of less frequent manufacturers is present, suggesting a highly skewed distribution where a few makes account for most accident records.

In [115]:
# check any issues

df1['Make'].unique()[:20]

array(['Boeing', 'Piper', 'Cessna', 'Beech', 'Bellanca', 'Grumman',
       'Schweizer', 'Mooney', 'Eckstein', 'Air Tractor', 'Taylorcraft',
       'Ayers', 'William F. Lair', 'American Champion (acac)', 'Ayres',
       'Rockwell', 'Aeronca', 'Bolger Scott', 'Aviat', 'Glenn L Smith'],
      dtype=object)

In [116]:
#cleaning tasks

df1['Make'] = df1['Make'].str.strip().str.upper()

df1['Make']

C:\Users\IBRAH\AppData\Local\Temp\ipykernel_29060\3869824205.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Make'] = df1['Make'].str.strip().str.upper()


4150                         BOEING
4171                          PIPER
8865                          PIPER
10605                        CESSNA
10688                        CESSNA
                    ...            
88846                         BEECH
88857                         BEECH
88865                        CESSNA
88873            CIRRUS DESIGN CORP
88886    AMERICAN CHAMPION AIRCRAFT
Name: Make, Length: 10086, dtype: object

### Obsevations

- The Make column contains aircraft manufacturers such as BOEING, PIPER, CESSNA, and BEECH, which dominate the accident dataset.

- CESSNA and PIPER appear most frequently, indicating that small general aviation aircraft account for a large proportion of recorded incidents.

- Some manufacturers are represented with slight naming variations (e.g., CIRRUS DESIGN CORP vs CIRRUS), requiring standardization for accurate grouping.

- The distribution is highly skewed, with a few manufacturers accounting for most accident records, while many others appear infrequently.

In [117]:
# handling missing values

df1 = df1.dropna(subset=['Make'])

df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est,Fatal_Injury_Rate,Serious_Injury_Rate,Severe_Injury_Rate,Is_Destroyed
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,588.0,VMC,Taxi,Probable Cause,04-12-2014,588.0,0.0,0.0,0.0,False
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,0.0,IMC,Cruise,Probable Cause,02-05-2011,2.0,0.5,0.5,1.0,True
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,15-12-2009,1.0,1.0,0.0,1.0,True
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,0.0,VMC,Maneuvering,Probable Cause,17-10-2016,1.0,1.0,0.0,1.0,True
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,12-01-2016,2.0,0.0,0.5,0.5,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,0.0,VMC,NaN,NaN,13-12-2022,3.0,1.0,0.0,1.0,True
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,2.0,VMC,NaN,NaN,28-12-2022,2.0,0.0,0.0,0.0,False
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,1.0,VMC,NaN,NaN,15-12-2022,1.0,0.0,0.0,0.0,False
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,1.0,VMC,NaN,NaN,27-12-2022,1.0,0.0,0.0,0.0,False


### Observations

- After data cleaning and feature engineering, the dataset contains 10,086 records with 36 variables, indicating successful filtering of relevant aviation accident cases.

- Key engineered safety metrics such as Fatal_Injury_Rate, Serious_Injury_Rate, Severe_Injury_Rate, and Is_Destroyed contain no missing values, ensuring reliability in outcome-based analysis.

- Operational and contextual variables such as Weather.Condition and Broad.phase.of.flight still contain some missing values, likely due to incomplete reporting during accident documentation.

- Overall, the dataset shows improved completeness in critical analytical variables, while remaining missing values are concentrated in secondary descriptive fields that are less critical to safety outcome modeling.

In [118]:
#keeping reasonable number of manufactures

make_counts = df1['Make'].value_counts()

valid_makes = make_counts[make_counts >= 50].index

df1 = df1[df1['Make'].isin(valid_makes)]

df1


,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est,Fatal_Injury_Rate,Serious_Injury_Rate,Severe_Injury_Rate,Is_Destroyed
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,588.0,VMC,Taxi,Probable Cause,04-12-2014,588.0,0.0,0.0,0.0,False
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,0.0,IMC,Cruise,Probable Cause,02-05-2011,2.0,0.5,0.5,1.0,True
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,15-12-2009,1.0,1.0,0.0,1.0,True
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,0.0,VMC,Maneuvering,Probable Cause,17-10-2016,1.0,1.0,0.0,1.0,True
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,12-01-2016,2.0,0.0,0.5,0.5,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,0.0,VMC,NaN,NaN,13-12-2022,3.0,1.0,0.0,1.0,True
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,2.0,VMC,NaN,NaN,28-12-2022,2.0,0.0,0.0,0.0,False
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,1.0,VMC,NaN,NaN,15-12-2022,1.0,0.0,0.0,0.0,False
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,1.0,VMC,NaN,NaN,27-12-2022,1.0,0.0,0.0,0.0,False


### Observations

- After cleaning and filtering the dataset, the number of records was reduced to 9,467 entries, focusing only on aircraft makes with sufficient representation for statistical analysis.

- Only manufacturers with a reasonable frequency (e.g., ≥50 occurrences) were retained to ensure reliability in safety comparisons and reduce bias from rare aircraft types.

- Major general aviation manufacturers such as CESSNA, PIPER, BEECH, and BOEING remain dominant in the filtered dataset, ensuring meaningful industry-level insights.

- This filtering step improves statistical robustness by ensuring that safety conclusions are drawn from adequately sampled aircraft makes rather than rare or outlier models.

### Inspect Model column
- Get rid of any NaNs.
- Inspect the column and counts for each model/make. Are model labels unique to each make?
- If not, create a derived column that is a unique identifier for a given plane type.

In [119]:
#removing NaNs

df1 = df1.dropna(subset=['Model'])

df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est,Fatal_Injury_Rate,Serious_Injury_Rate,Severe_Injury_Rate,Is_Destroyed
4150,20001214X42478,Incident,LAX83IA149A,1983-03-18,"LOS ANGELES, CA",United States,NaN,NaN,LAX,LOS ANGELES INTL,...,588.0,VMC,Taxi,Probable Cause,04-12-2014,588.0,0.0,0.0,0.0,False
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,0.0,IMC,Cruise,Probable Cause,02-05-2011,2.0,0.5,0.5,1.0,True
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,15-12-2009,1.0,1.0,0.0,1.0,True
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,0.0,VMC,Maneuvering,Probable Cause,17-10-2016,1.0,1.0,0.0,1.0,True
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,0.0,VMC,Maneuvering,Probable Cause,12-01-2016,2.0,0.0,0.5,0.5,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88846,20221206106419,Accident,CEN23FA060,2022-12-05,"Yukon, OK",United States,352933N,0974927W,RCE,CLARENCE E PAGE MUNI,...,0.0,VMC,NaN,NaN,13-12-2022,3.0,1.0,0.0,1.0,True
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,2.0,VMC,NaN,NaN,28-12-2022,2.0,0.0,0.0,0.0,False
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,1.0,VMC,NaN,NaN,15-12-2022,1.0,0.0,0.0,0.0,False
88873,20221215106463,Accident,ERA23LA090,2022-12-14,"San Juan, PR",United States,182724N,0066554W,SIG,FERNANDO LUIS RIBAS DOMINICCI,...,1.0,VMC,NaN,NaN,27-12-2022,1.0,0.0,0.0,0.0,False


In [120]:
#checking how models are distributed

df1.groupby('Make')['Model'].nunique().sort_values(ascending=False).head(10)

Make
CESSNA                        58
PIPER                         36
BEECH                          9
AMERICAN CHAMPION AIRCRAFT     6
CHAMPION                       6
BELLANCA                       5
BOEING                         5
MOONEY                         5
AIR TRACTOR                    4
AIR TRACTOR INC                3
Name: Model, dtype: int64

In [121]:
#checking overall model counts
df1['Model'].value_counts().head(20)

Model
172          745
152          308
182          299
172S         267
PA28         263
172N         244
SR22         232
180          213
172M         178
PA-18-150    175
150          172
PA-28-140    166
A36          162
172P         141
140          115
737          114
170B         107
172R         106
PA-28-180    105
PA-28-161    102
Name: count, dtype: int64

In [122]:
#creating a unique identifier

df1['Make_Model'] = df1['Make'].str.upper().str.strip() + "_" + df1['Model'].str.upper().str.strip()
df1['Make_Model']

4150                           BOEING_747
4171                      PIPER_PA-28-140
8865                      PIPER_PA-18-150
10605                         CESSNA_182A
10688                         CESSNA_182Q
                       ...               
88846                          BEECH_F33A
88857                           BEECH_A36
88865                          CESSNA_172
88873             CIRRUS DESIGN CORP_SR22
88886    AMERICAN CHAMPION AIRCRAFT_8GCBC
Name: Make_Model, Length: 9467, dtype: object

In [123]:
#filter low frequency aircraft

model_counts = df1['Make_Model'].value_counts()

valid_models = model_counts[model_counts >= 50].index

df1 = df1[df1['Make_Model'].isin(valid_models)]

df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est,Fatal_Injury_Rate,Serious_Injury_Rate,Severe_Injury_Rate,Is_Destroyed,Make_Model
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,IMC,Cruise,Probable Cause,02-05-2011,2.0,0.500000,0.500000,1.000000,True,PIPER_PA-28-140
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,VMC,Maneuvering,Probable Cause,15-12-2009,1.0,1.000000,0.000000,1.000000,True,PIPER_PA-18-150
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,VMC,Maneuvering,Probable Cause,17-10-2016,1.0,1.000000,0.000000,1.000000,True,CESSNA_182A
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,VMC,Maneuvering,Probable Cause,12-01-2016,2.0,0.000000,0.500000,0.500000,True,CESSNA_182Q
11898,20001214X36887,Accident,NYC85FA145B,1985-06-11,"BELMAR, NJ",United States,NaN,NaN,BLM,BELMAR MONMOUTH CO.,...,VMC,Takeoff,Probable Cause,08-04-2013,6.0,0.166667,0.166667,0.333333,True,CESSNA_152
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88839,20221205106409,Accident,ERA23FA079,2022-12-03,"Venice, FL",United States,027347N,0082272W,KVNC,Venice Municipal Airport,...,VMC,NaN,NaN,20-12-2022,3.0,1.000000,0.000000,1.000000,False,PIPER_PA28
88844,20221206106424,Accident,WPR23LA054,2022-12-04,"Fergus Falls, MN",United States,046936N,0096524W,FFM,Fergus Falls Municipal Airport,...,VMC,NaN,NaN,13-12-2022,2.0,0.000000,0.000000,0.000000,False,CESSNA_A185F
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,VMC,NaN,NaN,28-12-2022,2.0,0.000000,0.000000,0.000000,False,BEECH_A36
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,VMC,NaN,NaN,15-12-2022,1.0,0.000000,0.000000,0.000000,False,CESSNA_172


### Cleaning other columns
- there are other columns containing data that might be related to the outcome of an accident. We list a few here:
- Engine.Type
- Weather.Condition
- Number.of.Engines
- Purpose.of.flight
- Broad.phase.of.flight

Inspect and identify potential cleaning tasks in each of the above columns. Execute those cleaning tasks. 

**Note**: You do not necessarily need to impute or drop NaNs here.

In [124]:
#inspecting all columns

columns = [
    'Engine.Type',
    'Weather.Condition',
    'Number.of.Engines',
    'Purpose.of.flight',
    'Broad.phase.of.flight'
]

for col in columns:
    print(df1[col].value_counts(dropna=False).head())
    print("\n")

Engine.Type
Reciprocating    5676
NaN               880
Turbo Prop        122
Turbo Fan          18
Unknown             5
Name: count, dtype: int64


Weather.Condition
VMC    5873
NaN     521
IMC     263
Unk      43
UNK       2
Name: count, dtype: int64


Number.of.Engines
1.0    6133
NaN     400
2.0     169
Name: count, dtype: int64


Purpose.of.flight
Personal         4156
Instructional    1531
NaN               498
Business          105
Unknown            80
Name: count, dtype: int64


Broad.phase.of.flight
NaN         5670
Landing      515
Takeoff      167
Cruise       103
Approach      80
Name: count, dtype: int64




In [125]:
#cleaning task engine type

df1.loc[:, 'Engine.Type'] = df1['Engine.Type'].str.strip().str.title()
df1.loc[:, 'Engine.Type'] = df1['Engine.Type'].fillna('Unknown')

df1.loc[:, 'Engine.Type']

4171     Reciprocating
8865     Reciprocating
10605    Reciprocating
10688    Reciprocating
11898    Reciprocating
             ...      
88839          Unknown
88844          Unknown
88857          Unknown
88865          Unknown
88873          Unknown
Name: Engine.Type, Length: 6702, dtype: object

### Observations

- The Engine.Type column was cleaned by standardizing text formatting to ensure consistency in engine classification labels.

- Missing values were replaced with “Unknown” to retain records while preserving analytical integrity.

- The majority of aircraft in the dataset use Reciprocating engines, indicating that general aviation aircraft dominate the sample.

- The presence of a significant "Unknown" category suggests incomplete reporting in some accident records, which may slightly limit engine-type-specific conclusions.

In [126]:
#weather condition

df1.loc[:, 'Weather.Condition'] = df1['Weather.Condition'].str.strip().str.upper()
df1.loc[:, 'Weather.Condition'] = df1['Weather.Condition'].fillna('UNKNOWN')

df1.loc[:, 'Weather.Condition']

4171     IMC
8865     VMC
10605    VMC
10688    VMC
11898    VMC
        ... 
88839    VMC
88844    VMC
88857    VMC
88865    VMC
88873    VMC
Name: Weather.Condition, Length: 6702, dtype: object

### Observations

- The Weather.Condition column shows that the majority of recorded accidents occurred under VMC (Visual Meteorological Conditions), indicating clear weather conditions.

- A smaller proportion of accidents occurred under IMC (Instrument Meteorological Conditions), which typically represent low visibility or adverse weather conditions.

- The dominance of VMC suggests that many accidents are not solely driven by poor weather, but may be influenced by operational or human factors.

- This variable is important for assessing environmental risk, as comparing VMC and IMC conditions helps identify whether weather significantly contributes to accident severity.

In [127]:
#number of engines

df1.loc[:, 'Number.of.Engines'] = df1['Number.of.Engines'].fillna(df1['Number.of.Engines'].median())
df1.loc[:, 'Number.of.Engines'] = df1['Number.of.Engines'].astype(int)

df1.loc[:, 'Number.of.Engines']


4171     1.0
8865     1.0
10605    1.0
10688    1.0
11898    1.0
        ... 
88839    1.0
88844    1.0
88857    1.0
88865    1.0
88873    1.0
Name: Number.of.Engines, Length: 6702, dtype: float64

### Observations

- The Number.of.Engines column indicates the engine configuration of aircraft involved in accidents, with values predominantly concentrated around single-engine aircraft.

- The dataset shows that most recorded aircraft have 1 engine, suggesting that small general aviation aircraft make up a large proportion of accident records.

- The consistency of values (mostly 1.0) indicates limited variability in engine count within this filtered dataset, which may influence comparative analysis across aircraft types.

- This variable is useful for distinguishing between small and large aircraft categories, supporting the required segmentation in safety analysis.

In [128]:
#purpose of light

df1.loc[:,'Purpose.of.flight'] = df1['Purpose.of.flight'].str.strip().str.title()
df1.loc[:, 'Purpose.of.flight'] = df1['Purpose.of.flight'].fillna('Unknown')

df1.loc[:, 'Purpose.of.flight']


4171               Personal
8865     Aerial Application
10605             Skydiving
10688              Personal
11898         Instructional
                ...        
88839              Personal
88844              Personal
88857              Personal
88865         Instructional
88873              Personal
Name: Purpose.of.flight, Length: 6702, dtype: object

### Observations

- The Purpose.of.flight column shows the intended use of the aircraft during each accident, with common categories including Personal, Instructional, Skydiving, and Aerial Application.

- Personal flights dominate the dataset, indicating that general aviation and private use account for a large proportion of recorded accidents.

- Instructional flights also appear frequently, suggesting that training operations contribute meaningfully to the dataset and may involve higher exposure to risk due to student pilots.

- The diversity of flight purposes highlights varying operational risk profiles, which can be analyzed further to determine how mission type influences accident likelihood.

In [129]:
#broad phase of light

df1.loc[:, 'Broad.phase.of.flight'] = df1['Broad.phase.of.flight'].str.strip().str.title()
df1.loc[:, 'Broad.phase.of.flight'] = df1['Broad.phase.of.flight'].fillna('Unknown')

df1.loc[:, 'Broad.phase.of.flight']

4171          Cruise
8865     Maneuvering
10605    Maneuvering
10688    Maneuvering
11898        Takeoff
            ...     
88839        Unknown
88844        Unknown
88857        Unknown
88865        Unknown
88873        Unknown
Name: Broad.phase.of.flight, Length: 6702, dtype: object

### Observations

- The Broad.phase.of.flight column captures the stage of flight during which an accident occurred, with values such as Cruise, Takeoff, Maneuvering, and Landing appearing in the dataset.

- A significant number of records are labeled as "Unknown", indicating incomplete reporting or missing classification of flight phase in some accident records.

- Maneuvering and Cruise phases appear frequently in the dataset, suggesting that a notable proportion of accidents occur during in-flight operations rather than ground operations.

- The presence of diverse flight phases provides useful insight for identifying high-risk stages of flight in subsequent safety analysis.

### Column Removal
- inspect the dataframe and drop any columns that have too many NaNs

In [130]:
#inspecting missing values in all columns

(df1.isna().mean() * 100).sort_values(ascending=False)

Schedule                  95.180543
Air.carrier               55.565503
Airport.Code              28.245300
Airport.Name              28.111012
Report.Status             16.547299
Longitude                  5.117875
Latitude                   5.102954
Publication.Date           2.924500
FAR.Description            0.671441
Registration.Number        0.373023
Location                   0.014921
Weather.Condition          0.000000
Total.Serious.Injuries     0.000000
Total.Minor.Injuries       0.000000
Total.Uninjured            0.000000
Event.Id                   0.000000
Broad.phase.of.flight      0.000000
Total.Fatal.Injuries       0.000000
Fatal_Injury_Rate          0.000000
Serious_Injury_Rate        0.000000
Severe_Injury_Rate         0.000000
Is_Destroyed               0.000000
Total_Passengers_Est       0.000000
Engine.Type                0.000000
Purpose.of.flight          0.000000
Investigation.Type         0.000000
Number.of.Engines          0.000000
Amateur.Built              0

### Observations

- The missing value analysis shows that several columns contain significant proportions of missing data, with Schedule (95.18%), Air.carrier (55.57%), and Airport-related fields exceeding 25% missing values.

- A large set of key analytical variables, including injury metrics, aircraft classification, and engineered features, have 0% missing values, indicating successful prior cleaning and imputation.

- Location-based and reporting-related fields show moderate missingness, suggesting incomplete reporting or recording inconsistencies in certain accident records.

- The distribution of missing values supports the decision to retain only high-completeness variables for modeling, while excluding or down-weighting heavily missing features.

In [131]:
#dropping columns with too many NaNs

missing_value = df1.isna().mean()

cols_to_drop = missing_value[missing_value > 0.7].index

df1 = df1.drop(columns=cols_to_drop)

df1

,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date,Total_Passengers_Est,Fatal_Injury_Rate,Serious_Injury_Rate,Severe_Injury_Rate,Is_Destroyed,Make_Model
4171,20001214X42331,Accident,ATL83FA140,1983-03-20,"CROSSVILLE, TN",United States,NaN,NaN,NaN,NaN,...,IMC,Cruise,Probable Cause,02-05-2011,2.0,0.500000,0.500000,1.000000,True,PIPER_PA-28-140
8865,20001214X40407,Accident,MKC84FA197,1984-07-03,"WRIGHT, AR",United States,NaN,NaN,NaN,NaN,...,VMC,Maneuvering,Probable Cause,15-12-2009,1.0,1.000000,0.000000,1.000000,True,PIPER_PA-18-150
10605,20001214X41706,Accident,ATL85FA072,1984-12-30,"DUBLIN, VA",United States,NaN,NaN,PSK,NEW RIVER VALLEY,...,VMC,Maneuvering,Probable Cause,17-10-2016,1.0,1.000000,0.000000,1.000000,True,CESSNA_182A
10688,20001214X35509,Accident,DEN85LA064,1985-01-14,"WAPITI, WY",United States,NaN,NaN,NaN,NaN,...,VMC,Maneuvering,Probable Cause,12-01-2016,2.0,0.000000,0.500000,0.500000,True,CESSNA_182Q
11898,20001214X36887,Accident,NYC85FA145B,1985-06-11,"BELMAR, NJ",United States,NaN,NaN,BLM,BELMAR MONMOUTH CO.,...,VMC,Takeoff,Probable Cause,08-04-2013,6.0,0.166667,0.166667,0.333333,True,CESSNA_152
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
88839,20221205106409,Accident,ERA23FA079,2022-12-03,"Venice, FL",United States,027347N,0082272W,KVNC,Venice Municipal Airport,...,VMC,Unknown,NaN,20-12-2022,3.0,1.000000,0.000000,1.000000,False,PIPER_PA28
88844,20221206106424,Accident,WPR23LA054,2022-12-04,"Fergus Falls, MN",United States,046936N,0096524W,FFM,Fergus Falls Municipal Airport,...,VMC,Unknown,NaN,13-12-2022,2.0,0.000000,0.000000,0.000000,False,CESSNA_A185F
88857,20221213106447,Accident,ERA23LA086,2022-12-08,"Covington, GA",United States,332754N,0835049W,NaN,NaN,...,VMC,Unknown,NaN,28-12-2022,2.0,0.000000,0.000000,0.000000,False,BEECH_A36
88865,20221212106444,Accident,ERA23LA085,2022-12-12,"Knoxville, TN",United States,355745N,0835218W,DKX,KNOXVILLE DOWNTOWN ISLAND,...,VMC,Unknown,NaN,15-12-2022,1.0,0.000000,0.000000,0.000000,False,CESSNA_172


### Observation

- Columns with a high proportion of missing values were identified and removed to improve overall dataset quality and reliability.

- This step ensures that only features with sufficient data coverage are retained for analysis, reducing noise and potential bias in the dataset.

- The removal of sparsely populated columns improves computational efficiency and ensures more stable statistical analysis across aircraft records.

- After this cleaning step, the dataset contains 6,702 records and a reduced set of meaningful variables suitable for aircraft safety modeling and comparison.

In [132]:
#checking dropped cols

print(cols_to_drop)


Index(['Schedule'], dtype='object')


### Save DataFrame to csv
- its generally useful to save data to file/server after its in a sufficiently cleaned or intermediate state
- the data can then be loaded directly in another notebook for further analysis
- this helps keep your notebooks and workflow readable, clean and modularized

### Observation

- The cleaned dataset was exported to a CSV file to preserve the processed state of the data for future use.

- Saving the dataset allows for modular workflow separation, enabling exploratory data analysis and modeling to be performed independently from data cleaning.

- This approach improves reproducibility and efficiency by eliminating the need to re-run cleaning steps in future sessions.

- The saved dataset represents a fully processed and analysis-ready version of the aviation accident data.

In [133]:
#saving dataframe to csv

df1.to_csv('cleaned_aviation_data.csv', index=False)